### PR TITLE
CMake, CI, and Tests

### PR DESCRIPTION
**Pull request by kgerheiser**
Same old thing I've been doing.

Builds and runs:

https://github.com/kgerheiser/NCEPLIBS-g2/runs/751492737?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    - name: build-jasper
+      if: steps.cache-jasper.outputs.cache-hit != 'true'
+      run: |
+        cd jasper
+        mkdir cmake_build
+        cd build
+        cmake .. -DCMAKE_INSTALL_PREFIX=~/Jasper
+        make -j2
+        make install
```
Comment (aerorahul) [inline .github/workflows/main.yml:40]: any reason why Jasper cannot be treated as png, zlib and jpeg?

Comment (aerorahul) [reply .github/workflows/main.yml:40]: Is there a particular minimum version of Jasper that is necessary? If so, that should be documented.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (aerorahul) [inline README.md:1]: merge artifact.
please remove.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -1,71 +1,86 @@
...
+    "${fortran_${kind}_flags}")
+  set_target_properties(${lib_name}_f PROPERTIES Fortran_MODULE_DIRECTORY
+    ${module_dir})
+
+  add_library(${lib_name}_c OBJECT ${c_src})
+  target_include_directories(${lib_name}_c
+    PRIVATE
+    ${ZLIB_INCLUDE_DIRS}
+    ${PNG_INCLUDE_DIRS}
+    ${JASPER_INCLUDE_DIR})
```
Comment (aerorahul) [inline src/CMakeLists.txt:40]: you never did a `find_package(Jasper)`

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -1,71 +1,86 @@
...
+  set_target_properties(${lib_name}_f PROPERTIES COMPILE_FLAGS
+    "${fortran_${kind}_flags}")
+  set_target_properties(${lib_name}_f PROPERTIES Fortran_MODULE_DIRECTORY
+    ${module_dir})
+
+  add_library(${lib_name}_c OBJECT ${c_src})
+  target_include_directories(${lib_name}_c
+    PRIVATE
+    ${ZLIB_INCLUDE_DIRS}
+    ${PNG_INCLUDE_DIRS}
```
Comment (aerorahul) [inline src/CMakeLists.txt:39]: png supports namespace.  you could just do `png::png` in `target_link_libraries`.

Comment (aerorahul) [reply src/CMakeLists.txt:39]: See what is being done in nceplibs-g2c

File: src/list_of_files.cmake [hunk 5]
```diff
@@ -0,0 +1,71 @@
+set(fortran_src
+  ${CMAKE_CURRENT_SOURCE_DIR}/addfield.f
```
Comment (aerorahul) [inline src/list_of_files.cmake:2]: `${CMAKE_CURRENT_SOURCE_DIR}` is not necessary.
Please remove.

File: CMakeLists.txt [hunk 6]
```diff
@@ -24,67 +26,10 @@ find_package(Jasper REQUIRED)
 find_package(PNG REQUIRED)
 find_package(ZLIB REQUIRED)
```
Comment (aerorahul) [inline CMakeLists.txt:28]: a `find_package(Jasper)` is missing here.

Comment (kgerheiser) [reply CMakeLists.txt:28]: It's there, right above the PNG line

File: src/list_of_files.cmake (hunk 5 above)
Comment (kgerheiser) [reply src/list_of_files.cmake:2]: Ah yes, I did this before I started removing them

File: src/CMakeLists.txt (hunk 4 above)
Comment (kgerheiser) [reply src/CMakeLists.txt:39]: Noted

File: src/CMakeLists.txt (hunk 3 above)
Comment (kgerheiser) [reply src/CMakeLists.txt:40]: It's there

https://github.com/kgerheiser/NCEPLIBS-g2/blob/87cd25ffc246f352d0070d8318356b1e1a7dc5a4/CMakeLists.txt#L25

File: README.md (hunk 2 above)
Comment (kgerheiser) [reply README.md:1]: Yes

File: .github/workflows/main.yml (hunk 1 above)
Comment (kgerheiser) [reply .github/workflows/main.yml:40]: zlib and png are installed by default on Github runners (and maybe Ubuntu as a whole), and are available as Ubuntu packages, Jasper isn't.

As far as version goes, I don't know. I've seen 1.9.0, 1.9.28, and 2.xx. I just picked the version that I found in NCEPLIBS-externals.

**Comment by kgerheiser** [comment]
Done

**Comment by BoiVuong-NOAA** [comment]
Looks Good !

**Pull request merged**